### PR TITLE
fa-636 add psc discrepancy type

### PR DESCRIPTION
**Pull request by ICraig-CH**
feature:
- added new variable to PscDiscrepancy, PscDiscrepancyEntityData
- altered tests to include new variable
- added variable to the mapper and mapper tests

jira ticket: https://companieshouse.atlassian.net/browse/FAML-636

### CODE REVIEW

File: src/main/java/uk/gov/ch/pscdiscrepanciesapi/models/rest/PscDiscrepancy.java [hunk 1]
```diff
@@ -40,36 +52,35 @@ public void setPscDateOfBirth(String pscDateOfBirth) {
     }
 
     @Override
-    public int hashCode() {
```
Comment (CH-JonathanBlake) [inline src/main/java/uk/gov/ch/pscdiscrepanciesapi/models/rest/PscDiscrepancy.java:43]: I don't think we meant to change these but I also don't think this hurts.

Comment (ICraig-CH) [reply src/main/java/uk/gov/ch/pscdiscrepanciesapi/models/rest/PscDiscrepancy.java:43]: I regenerated the hashcode/equals when we added the new variables, I think GitHub is displaying the code changes weirdly again

Comment (CH-JonathanBlake) [reply src/main/java/uk/gov/ch/pscdiscrepanciesapi/models/rest/PscDiscrepancy.java:43]: no, besides the reordering the new fields, the new and old methods are distinct, look at the new hashcode method for the most obvious example

Comment (ICraig-CH) [reply src/main/java/uk/gov/ch/pscdiscrepanciesapi/models/rest/PscDiscrepancy.java:43]: do you think I should change it or is it fine as is?

Comment (CH-JonathanBlake) [reply src/main/java/uk/gov/ch/pscdiscrepanciesapi/models/rest/PscDiscrepancy.java:43]: I think it is fine, this matches what we've done in docs.developer and applications.api better.
And Jo isn't around to get lead feedback. But we could still ping Titans to ask why they implemented it this way

**Review by CH-JonathanBlake — APPROVED** [review]

**Review by ch-rfishwick — APPROVED** [review]

**Pull request merged**